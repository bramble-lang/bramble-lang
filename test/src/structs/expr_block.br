fn my_main() {
    let t: MyStruct := if (true) {
        let x:i32 := 5;
        let y:i32 := 1;
        let z: i32 := -1;
        let ms0:MyStruct := MyStruct{x: 11, y: 22, z: 33};
        let ms1:MyStruct := MyStruct{x: 111, y: 222, z: 333};
        let ms2:MyStruct := MyStruct{x: x * ms0.x * ms1.x, y: y + ms0.y + ms1.y, z: z};
        ms2
    } else {
        MyStruct{x: 0, y: 0, z: 0}
    };

    printiln t.x;
    printiln t.y;
    printiln t.z;
    
    return;
}

struct MyStruct {
    x: i32,
    y: i32,
    z: i32,
}