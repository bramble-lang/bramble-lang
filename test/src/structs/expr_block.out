6105
245
-1
